### ISSUE TITLE
Crash on __NSSingleObjectArrayI objectAtIndex in PlotPSqLiteNotificationReadDao 

### ISSUE THREAD

**Issue by ddarren** [Plotprojects/plotprojects-react-native-module#2]
We are experiencing crashes on iOS. Not right way but after some time has gone by, the app will crash on startup.

Some background:

We just upgraded our PlotProjects to the new React Native package and are now seeing the below error on only iOS.
For iOS the previous Pod version of PlotPlugin was 3.3.0 and is now 3.4.6.

We also use react-native-push-notification 7.4.0 with @react-native-community/push-notification-ios 1.8.0 for local notifications, which we have also recently upgraded.

Full stack trace below.

My only hunch is that this package does not play well when you do your own UNUserNotificationCenter

Before we upgraded to use the plotprojects-react-native-module package, we followed the iOS setup instruction and made sure to put `[Plot initializeWithDelegate:self];` after `UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];` in ios/VisitWidget/AppDelegate.m as specified  [here](https://files.plotprojects.com/documentation/ios/3.4.3/how-to-guides/iOS-integration-guide/#step-6-implement-appdelegate_1)

The old way worked. Now since using the npm package plotprojects-react-native-module, we followed the package's instructions and the example project https://github.com/Plotprojects/plot-react-native-module-example and remove the Plot code from AppDelegate.m
Now what I'm wondering is if plotprojects-react-native-module is ensuring that the Plot initializeWithDelegate is called after UNUserNotificationCenter.

```
OS Version: iOS 14.6 (18F72)
Report Version: 104

Exception Type: EXC_CRASH (SIGABRT)
Crashed Thread: 23

Application Specific Information:
*** -[__NSSingleObjectArrayI objectAtIndex:]: index 1 beyond bounds [0 .. 0]

Thread 23 Crashed:
0   CoreFoundation                  0x31e0cf754         __exceptionPreprocess
1   libobjc.A.dylib                 0x34775e7a4         objc_exception_throw
2   CoreFoundation                  0x31dfcacd4         -[__NSSingleObjectArrayI objectAtIndex:]
3   PlotProjects                    0x101ece020         -[PlotPSqLiteNotificationReadDao replacePlaceholdersForNearestQuery:includeIds:]
4   PlotProjects                    0x101ecd400         -[PlotPSqLiteNotificationReadDao findNearestNotifications:count:includeOffered:regionType:includeIds:]
5   PlotProjects                    0x101e9b45c         -[PlotPGeofenceMatchingService checkForGeofenceMatches:andContext:andLocation:]
6   PlotProjects                    0x101e98b9c         -[PlotPFenceMatchService locationAcquired:backgroundTask:plotContext:]
7   PlotProjects                    0x101e9a940         -[PlotPLocationUpdateService deliverCurrentLocationToMatchService:]
8   PlotProjects                    0x101e9a644         -[PlotPLocationUpdateService nextStep:]
9   PlotProjects                    0x101ee4750         -[PlotPNotificationService loadNotificationsCompleted:publicToken:storeEvents:plotContext:error:]
10  PlotProjects                    0x101ee85bc         -[PlotPDataUpdateDao httpRequest:completedWithResponse:andError:andPlotContext:]
11  PlotProjects                    0x101ea5128         -[PlotPRawHttpRequest completedWithResponse:andError:]
12  PlotProjects                    0x101ea51b8         -[PlotPRawHttpRequest connectionDidFinishLoading:]
13  CFNetwork                       0x31f06fa9c         __CFTubeSetTubeTypeNotifier
14  CFNetwork                       0x31ee03ec8         <redacted>
15  CFNetwork                       0x31ee03f24         <redacted>
16  CFNetwork                       0x31ef97754         _CFNetworkSetATSContext
17  CFNetwork                       0x31ef9616c         _CFNetworkSetATSContext
18  libdispatch.dylib               0x31d9ce818         _dispatch_client_callout
19  libdispatch.dylib               0x31d9d2308         _dispatch_block_invoke_direct
20  CFNetwork                       0x31f04cab4         _CFURLStorageSessionCopyCache
21  CoreFoundation                  0x31dfdb87c         CFArrayApplyFunction
22  CFNetwork                       0x31f04c9c0         _CFURLStorageSessionCopyCache
23  CFNetwork                       0x31f04f140         _CFURLStorageSessionCopyCache
24  CoreFoundation                  0x31e0499e4         __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
25  CoreFoundation                  0x31e0498e0         __CFRunLoopDoSource0
26  CoreFoundation                  0x31e048be4         __CFRunLoopDoSources0
27  CoreFoundation                  0x31e042bc4         __CFRunLoopRun
28  CoreFoundation                  0x31e04235c         CFRunLoopRunSpecific
29  Foundation                      0x3206b1fd8         -[NSRunLoop(NSRunLoop) runMode:beforeDate:]
30  PlotProjects                    0x101ec5efc         -[PlotPSystemTimerDao runThread:]
31  Foundation                      0x320823328         __NSThread__start__
32  libsystem_pthread.dylib         0x3b58bbbf8         _pthread_start
33  libsystem_pthread.dylib         0x3b58c4754         thread_start
```

```
React Native info is:
System:
   OS: macOS 11.5
   CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
   Memory: 1.23 GB / 32.00 GB
   Shell: 5.8 - /bin/zsh
 Binaries:
   Node: 14.17.1 - ~/.nvm/versions/node/v14.17.1/bin/node
   Yarn: Not Found
   npm: 6.14.13 - ~/.nvm/versions/node/v14.17.1/bin/npm
   Watchman: 2021.06.07.00 - /usr/local/bin/watchman
 Managers:
   CocoaPods: Not Found
 SDKs:
   iOS SDK:
     Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
   Android SDK: Not Found
 IDEs:
   Android Studio: 4.2 AI-202.7660.26.42.7486908
   Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
 Languages:
   Java: 16.0.1 - /usr/bin/javac
 npmPackages:
   @react-native-community/cli: Not Found
   react: 17.0.2 => 17.0.2
   react-native: 0.64.2 => 0.64.2
   react-native-macos: Not Found
 npmGlobalPackages:
   *react-native*: Not Found
```

**Comment by yahya-plotprojects** [comment]
I can't reproduce the issue but it has nothing to do with the NotificationCenter. That part of the code is related to the geofence monitoring.
Can you please share the app container (please email it as it could contain sensitive info)?

**Comment by yahya-plotprojects** [comment]
This issue is now fixed in the new release (v1.0.4) of PlotProjects react native module. I will close this now.
